src/Microsoft.SqlTools.ServiceLayer/Hosting/Protocol/ProtocolEndpoint.cs

```diff
@@ -194,37 +203,52 @@ public void WaitForExit()
             EventType<TParams> eventType,
             TParams eventParams)
         {
-            if (!this.protocolChannel.IsConnected)
+            try
             {
-                throw new InvalidOperationException("SendEvent called when ProtocolChannel was not yet connected");
-            }
+                if (!this.protocolChannel.IsConnected)
+                {
+                    throw new InvalidOperationException("SendEvent called when ProtocolChannel was not yet connected");
+                }
 
-            // Some events could be raised from a different thread.
-            // To ensure that messages are written serially, dispatch
-            // dispatch the SendEvent call to the message loop thread.
+                // Some events could be raised from a different thread.
+                // To ensure that messages are written serially, dispatch
+                // dispatch the SendEvent call to the message loop thread.
 
-            if (!this.MessageDispatcher.InMessageLoopThread)
-            {
-                TaskCompletionSource<bool> writeTask = new TaskCompletionSource<bool>();
+                if (!this.MessageDispatcher.InMessageLoopThread)
+                {
+                    TaskCompletionSource<bool> writeTask = new TaskCompletionSource<bool>();
 
-                this.MessageDispatcher.SynchronizationContext.Post(
-                    async (obj) =>
-                    {
-                        await this.protocolChannel.MessageWriter.WriteEvent(
-                            eventType,
-                            eventParams);
+                    this.MessageDispatcher.SynchronizationContext.Post(
+                        async (obj) =>
+                        {
+                            await this.protocolChannel.MessageWriter.WriteEvent(
+                                eventType,
+                                eventParams);
 
-                        writeTask.SetResult(true);
-                    }, null);
+                            writeTask.SetResult(true);
+                        }, null);
 
-                return writeTask.Task;
+                    return writeTask.Task;
+                }
+                else
+                {
+                    return this.protocolChannel.MessageWriter.WriteEvent(
+                        eventType,
+                        eventParams);
+                }
             }
-            else
+            catch (Exception ex)
             {
-                return this.protocolChannel.MessageWriter.WriteEvent(
-                    eventType,
-                    eventParams);
+                if (SendEventIgnoreExceptions)
+                {
+                    Logger.Write(LogLevel.Verbose, "Exception in SendEvent " + ex.ToString());
+                }
+                else
+                {
+                    throw;
+                }
             }
+            return null;
```

Comment:
I think Task.FromResult(false) may be better to avoid nullrefs in other test scenarios